packages/fx-core/src/plugins/resource/appstudio/appStudio.ts

```diff
@@ -135,9 +136,13 @@ export namespace AppStudioClient {
           `Cannot get the app definition with app ID ${teamsAppId}, due to ${e.name}: ${e.message}`
         );
       }
-      throw new Error(
+      const err = new Error(
         `Cannot get the app definition with app ID ${teamsAppId}, due to ${e.name}: ${e.message}`
       );
+      if (e.response?.data?.status) {
+        err.name = e.response?.data?.status;
```

Comment:
Use name is confusing, how about use statusCode since error is any type?

Reply:
Error type only have name and message property. I am also considering whether it is better to add a new interface here.